### DISCUSSION TITLE
Image edit after upload

### DISCUSSION THREAD

**Discussion by meerzulee** [directus/directus#9643] [Archive]
**A feature editing the image after upload would be great.** 
I want to implement this feature on my own. 
Should I create an extension or just make changes and rebuild the project?

**Comment by Nitwel** [comment]
You could either modify the app itself to open the edit page when a image gets uploaded or add a new interface that after uploading the image, allows you to edit it.

**Comment by rijkvanzanten** [comment]
Another approach would be to use the file upload hook to asynchronously apply any transformations to the file on upload.

Could be a nice feature to force every file upload to to go through a pre-defined transformation before saving to disk, to enforce things like aspect-ratio or max dimensions. That being said, I'd personally recommend storing the origin files to disk, and relying on the /assets endpoint (or third party service) to fetch the thumbnails in various sizes. That way, you never lose access to the origin file in case you ever need to use a bigger version. (I really shot myself in the foot with this exact problem in previous client work 😓)

**Reply by Rar9** [reply]
Would it be easly possible to add focal point data.

**Reply by azrikahar** [reply]
> Would it be easly possible to add focal point data.

There's a feature request for focal point in #7901, so do give it an upvote 👍

#14578 may also indirectly make it easier (not the focal point logic, but the file version/reference portion), with an ongoing draft PR over here: #15634.
